docs/proposals/promotion-strategy.md

```diff
@@ -0,0 +1,475 @@
+  url: # For details about the status
+status:
+  plugin:
+    github:
+      checkRunId: # Optional int
+    gitlab:
+    # others can store their status here
+```
+
+#### PullRequest
+
+A PullRequest represents a PR in some SCM.
+
+For the purposes of environment promotion, this resource will be created by the ProposedCommit and RevertCommit controllers.
+
+```yaml
+apiVersion: promoter.argoproj.io/v1alpha1
+kind: PullRequest
+spec:
+  title:
+  description:
+  sourceBranch:
+  targetBranch:
+status:
+  plugin:
+    github:
+      id:
+    gitlab:
+      id:
+```
+### How to Use
+
+The GitOps Promoter, by itself, can't do much. An external tool has to provide information about active commits, proposed commits, and commit statuses.
+
+The external tools are responsible for the following:
+
+1) Creating a PromotionStrategy defining the order of environment promotion. This info will likely come from the end user.
+2) Updating the "-next" branch for an environment when a new hydrated manifest is available for promotion, and adding a git note to the commit to refer back to the corresponding dry commit SHA.
+3) Keeping all CommitStatus resources up to date for all active and proposed commits.
+
+### Use cases
+
+### Security Considerations
+
+This focus so far in this document has been user experience. We need to spend a _lot_ more time building out the specifics of the security considerations for this tool.
+
+#### Git Push Access
+
+This tool inherently must have git push access. We need to evaluate what systems/recommendations/documentation we need to make the system secure.
+
+At a minimum, no component should have access to git push secrets which does not _need_ that access. For example, the controller which opens GitHub PRs should live in its own namespace with access to its push secrets, but the Argo CD ApplicationSet controller (for example) should not have access to those secrets.
+
+We should also encourage users to use their SCM's branch protection features to limit environment branch push access to only the promotion/rollback tool. Developers shouldn't directly manipulate those branches.
+
+#### Secret-Injection CMPs
+
+This tool is not compatible with CMPs that inject secrets, because those secrets would then be pushed to git.
+
+This tool will not be responsible for checking that no secrets are being written to git - that's the responsibility of the hydrator tool is creating ProposedCommits (for our purposes, that's Argo CD).
+
+### Risks and Mitigations
+
+### Upgrade / Downgrade Strategy
+
+## Drawbacks
+
+## Alternatives
+
+* Kargo
+    * Relies on many new concepts (Freight, Warehouse, etc.) instead of existing git/SCM concepts - i.e. requires training for developers to use
+    * Currently focuses on promote-by-sync and promote-by-push (which have drawbacks, as described in [Motivation](#motivation))
+    * Could be adapted to implement promote-by-pr-merge, but this is just one mode of many
+* Telefonistka
```

Comment:
Some shameless promotion :)

```suggestion
* [Kargo](https://github.com/akuity/kargo)
    * Relies on many new concepts (Freight, Warehouse, etc.) instead of existing git/SCM concepts - i.e. requires training for developers to use
    * Currently focuses on promote-by-sync and promote-by-push (which have drawbacks, as described in [Motivation](#motivation))
    * Could be adapted to implement promote-by-pr-merge, but this is just one mode of many
* [Telefonistka](https://github.com/wayfair-incubator/telefonistka)
```

Reply:
We are in the business of promotion here ;-)